LIB_DOC.md

```diff
@@ -0,0 +1,246 @@
+Subxt is a library to **sub**mit e**xt**rinsics to a [substrate](https://github.com/paritytech/substrate) node via RPC.
```

Comment:
I'm not a big fan of the name LIB_DOC.md; if we'd like to split the readme up, let's add something like a `docs` folder and then put things in that, or otherwise let's keep the readme in one place and perhaps add a table of contents if it's getting a bit harder to navigate/find things :)